web/src/message_fetch.js

```diff
@@ -463,30 +403,31 @@ export function maybe_load_older_messages(opts) {
     // This function gets called when you scroll to the top
     // of your window, and you want to get messages older
     // than what the browsers originally fetched.
-    const msg_list = opts.msg_list;
-    if (!msg_list.data.fetch_status.can_load_older_messages()) {
+    const msg_list_data = opts.msg_list_data ?? opts.msg_list.data;
+    if (!msg_list_data.fetch_status.can_load_older_messages()) {
         // We may already be loading old messages or already
         // got the oldest one.
         return;
     }
 
     do_backfill({
-        msg_list,
+        ...opts,
         num_before: opts.recent_view
             ? consts.recent_view_fetch_more_batch_size
             : consts.narrowed_view_backward_batch_size,
     });
 }
 
 export function do_backfill(opts) {
-    const msg_list = opts.msg_list;
-    const anchor = get_backfill_anchor(msg_list);
+    const msg_list_data = opts.msg_list_data ?? opts.msg_list.data;
+    const anchor = get_backfill_anchor(msg_list_data);
 
     load_messages({
         anchor,
         num_before: opts.num_before,
         num_after: 0,
-        msg_list,
+        msg_list: opts.msg_list,
+        msg_list_data: opts.msg_list_data,
```

Comment:
Is this supposed to be the `msg_list_data` you computed above, not `opts.msg_list_data`?

Reply:
No, we want to call `load_messages` similarly to how `do_backfill` was called so that we know if we are filling messages in a message list or just the date structure. One of `msg_list` and `msg_list_data` will always be `undefined`.

Reply:
OK -- probably worth a comment or maybe a different variable name.

Reply:
Added.